### PR TITLE
CodeDom fixes for explicit interface implementation with built-in types

### PR DESCRIPTION
**Pull request by marek-safar**
This is quite corner case but C#/VB compilers cannot even parse code to show meaningful error message where built-in types are used as prefix in explicit interface implementation

The patch fixes cases like the one bellow to always use fully qualified name in this scenario.
```c#
public class Test1
{
    event int int.Click;
}
```

### CODE REVIEW

**Comment by stephentoub** [comment]
Thanks for the fix.  Could you add a test for this as well?

File: src/System.CodeDom/src/Microsoft/CSharp/CSharpCodeGenerator.cs [hunk 1]
```diff
@@ -2803,68 +2803,56 @@ public string CreateEscapedIdentifier(string name)
...
+                        return "ulong";
+                    case "system.sbyte":
+                        return "sbyte";
+                    case "system.single":
+                        return "float";
+                    case "system.double":
+                        return "double";
+                    case "system.decimal":
+                        return "decimal";
+                }
```
Comment (Priya91) [inline src/System.CodeDom/src/Microsoft/CSharp/CSharpCodeGenerator.cs]: The default case is moved out of switch statement, the last else part should be in default?

**Comment by Priya91** [comment]
@marek-safar Can you provide before and after behavior from this change? This will have to be ported back to desktop

File: src/System.CodeDom/src/Microsoft/CSharp/CSharpCodeGenerator.cs (hunk 1 above)
Comment (marek-safar) [inline src/System.CodeDom/src/Microsoft/CSharp/CSharpCodeGenerator.cs]: Why? Do you want me to duplicate the whole else block to default switch section?

Comment (stephentoub) [inline src/System.CodeDom/src/Microsoft/CSharp/CSharpCodeGenerator.cs]: Seems fine to me as-is.  @Priya91, what's the concern?

Comment (Priya91) [inline src/System.CodeDom/src/Microsoft/CSharp/CSharpCodeGenerator.cs]: The diff showed the default case been removed. Saw the full file now, makes sense. The diff was confusing.

**Comment by marek-safar** [comment]
@stephentoub @Priya91 I've added simple test for the change

**Comment by stephentoub** [comment]
> I've added simple test for the change

Thanks for adding it, @marek-safar.  The src change, though, adds over 15 special cases, and the test only covers one.  The change also applies to both C# and VB, but the test only covers C#.  Could you expand it accordingly?

**Comment by stephentoub** [comment]
Replaced by https://github.com/dotnet/corefx/pull/15918

**Pull request closed without merging**